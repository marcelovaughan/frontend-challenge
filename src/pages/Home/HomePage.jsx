import React from 'react'

import { Layout } from 'components'
import { Home } from 'views'

import './App.module.css'

const HomePage = () => (
  <Layout>
    <Home />
  </Layout>
)

export default HomePage
