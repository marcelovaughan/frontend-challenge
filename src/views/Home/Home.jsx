import React from 'react'
import { Button } from 'plurall-ui'
import { Link } from 'react-router-dom'

import { SubHeader } from 'components'
import { SomosClient } from 'utils'

import styles from './Home.module.css'

class Home extends React.Component {
  state = {}

  client = new SomosClient()

  render() {
    return (
      <React.Fragment>
        <SubHeader
          breadcrumb={[{ text: 'Home' }]}
          heading="Somos Front-end Challange"
        />
        <div className={styles.wrapper}>
          <Link to="/busca"><Button>Encontre um artista</Button></Link>
        </div>
      </React.Fragment>
    )
  }
}

export default Home
