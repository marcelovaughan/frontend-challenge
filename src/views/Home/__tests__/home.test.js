import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { configure, shallow, mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"
import { shape } from 'prop-types';

import Home from '../Home';

configure({ adapter: new Adapter() });

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
};

const createContext = () => ({
    context: { router },
    childContextTypes: { router: shape({}) },
  });

function shallowWrap(node) {
  return shallow(node, createContext());
}

describe('Home', () => {  
  const wrappedShallow = () => shallowWrap(<Home />);
  test('should render home', () => {
    const wrapper = wrappedShallow();
    expect(wrapper).toMatchSnapshot();
  });
});