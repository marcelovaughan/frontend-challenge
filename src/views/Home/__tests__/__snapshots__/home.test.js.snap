// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render home 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SubHeader
          breadcrumb={
            Array [
              Object {
                "text": "Home",
              },
            ]
          }
          heading="Somos Front-end Challange"
        />,
        <div
          className="wrapper"
        >
          <Link
            replace={false}
            to="/busca"
          >
            <Button
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Encontre um artista
            </Button>
          </Link>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breadcrumb": Array [
            Object {
              "text": "Home",
            },
          ],
          "heading": "Somos Front-end Challange",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/busca"
          >
            <Button
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Encontre um artista
            </Button>
          </Link>,
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Encontre um artista
            </Button>,
            "replace": false,
            "to": "/busca",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Encontre um artista",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Encontre um artista",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SubHeader
            breadcrumb={
              Array [
                Object {
                  "text": "Home",
                },
              ]
            }
            heading="Somos Front-end Challange"
          />,
          <div
            className="wrapper"
          >
            <Link
              replace={false}
              to="/busca"
            >
              <Button
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Encontre um artista
              </Button>
            </Link>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumb": Array [
              Object {
                "text": "Home",
              },
            ],
            "heading": "Somos Front-end Challange",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/busca"
            >
              <Button
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Encontre um artista
              </Button>
            </Link>,
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Encontre um artista
              </Button>,
              "replace": false,
              "to": "/busca",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Encontre um artista",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Encontre um artista",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
